### ISSUE TITLE
The running code of singularity container is inconsistent with CUDA

### ISSUE THREAD

**Issue by davidyang180** [sylabs/singularity#746]
Hi! When I run Python code on the command line, CUDA showing Python is available; 
![image](https://user-images.githubusercontent.com/45596935/165526837-1af19261-01cb-4090-bd75-f9c251daf27f.png)
But when I run the code through the configuration INI file, torch.exe will appear_ C._ cuda_ getDeviceCount() = 0，torch. cuda. is_ In the case of available () = false, what is the reason?

![image](https://user-images.githubusercontent.com/45596935/165527977-6d4c9dd4-b4cf-4e16-9953-9368f195ad22.png)
![image](https://user-images.githubusercontent.com/45596935/165527697-70ba5982-be7d-4e86-b810-dacca72828b2.png)

**Comment by dtrudg** [comment]
Hello. It's not clear to me what you mean by "when I run the code through the configuration INI file". What application is using this ini file? How are you running the application? You'll need to provide more information about the worfklwo and container in use here. 

Also note that the Slack channel and Google Group are better venues for application-specific issues. Given that CUDA works from a python prompt this is not likely to be an issue with SingularityCE itself.

https://sylabs.io/singularity#community

**Comment by davidyang180** [comment]
> Hello. It's not clear to me what you mean by "when I run the code through the configuration INI file". What application is using this ini file? How are you running the application? You'll need to provide more information about the worfklwo and container in use here.
> 
> Also note that the Slack channel and Google Group are better venues for application-specific issues. Given that CUDA works from a python prompt this is not likely to be an issue with SingularityCE itself.
> 
> https://sylabs.io/singularity#community
Hi! Thanks for your reply. This is the official code evaluation method of VOT Challenge. By configuring the ini file, call the python command to run; For example, configure the following "trackers.ini“ file，then call the configuration file through the official package of VOT Challenge to run the python code.

the "trackers.ini“ file:
``
[parking_point_adaptive]  # <tracker-name>
label = APMT
protocol = traxpython
command =import vot_test_toolkit.run_vot as run_vot; run_vot.run_vot('parking_point_adaptive', 'baseline_motion')

# Specify a path to trax python wrapper if it is not visible (separate by ; if using multiple paths)
paths = /Projects/APMT_motion/
``
The problem is that CUDA is available by running Python code on the command line; However, CUDA cannot be used by calling Python command in this way.

**Comment by dtrudg** [comment]
It's still not clear what tool / workflow / application this is, or how you are running the code that operates on this ini file.

If it works from the command line, then Singularity is not causing the problem here. You will need to find the issue in the way that the tool is calling Python. Perhaps it is overriding the `LD_LIBRARY_PATH` that Singularity sets when using `--nv` to bind the CUDA drivers.

I'm afraid that this is an application issue, not a Singularity problem - so it is best to raise it in the support locations for the project that you are using.

**Comment by davidyang180** [comment]
> It's still not clear what tool / workflow / application this is, or how you are running the code that operates on this ini file. I'm not familiar with what the VOT challenge is, or what software it uses.
> 
> If it works from the command line, then Singularity is not causing the problem here. You will need to find the issue in the way that the tool is calling Python. Perhaps it is overriding the `LD_LIBRARY_PATH` that Singularity sets when using `--nv` to bind the CUDA drivers.
> 
> I'm afraid that this is an application issue, not a Singularity problem - so it is best to raise it in the support locations for the project that you are using.

Tranks! Another question is, when I use the singularity command to run code, 
```
singularity exec --nv <sandboxFile-path> <command> or singularity exec --nv <sif-path> <command>
```
I have set the sandboxFile and SIF file to fakeroot, what causes this error?
```
[Errno 30] Read-only file system: 
```

**Comment by dtrudg** [comment]
You need to use `--writable` with the sandbox, if the program expects to write files to the container.

The SIF is read-only. You can use `--writable-tmpfs` to allow changes to be written temporarily, that will be lost on exit. Otherwise see the overlays section in the user guide at https://sylabs.io/docs

**Comment by davidyang180** [comment]
> You need to use `--writable` with the sandbox, if the program expects to write files to the container.
> 
> The SIF is read-only. You can use `--writable-tmpfs` to allow changes to be written temporarily, that will be lost on exit. Otherwise see the overlays section in the user guide at https://sylabs.io/docs

Yes! I used this command but it gave me another error. Is this because of the problem of mounting the local directory when creating the container, how should I solve it?
```
WARNING: nv files may not be bound with --writable
WARNING: By using --writable, Singularity can't create /data destination automatically without overlay or underlay
FATAL:   container creation failed: mount /usr/local/var/singularity/mnt/session/data->/data error: while mounting /usr/local/var/singularity/mnt/session/data: destination /data doesn't exist in container
```
